File: problems/intermediate/secret_agent/__tests__/secret agent.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "../intermediate/secret agent/secret agent.c"

void test(){
    bool passed = true;
    if (crackCode("0")!=["0", "8"])
    {
        printf("❌ failed c test1 ");
        passed = false;
    }
    if (crackCode("2")!= ["1", "2", "3", "5"])
    {
        printf("❌ failed c test2 ");
        passed = false;
    }
    if (passed)
    {
        printf("✅ passed c tests");
    }
    
    

}

int main (){
    test();
    return 0;
}

File: problems/intermediate/secret_agent/__tests__/secret agent.py
import sys
import os

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from secret_agent import crackCodeN

def test():
    if crackCodeN("0") != ['0', '8']:
        print("❌ failed python test 1")
    if crackCodeN("2") != ['1', '2', '3', '5']:
        print("❌ failed python test 1")
    else:
        print("✅ passed all python tests")


if __name__ == "__main__":
    test()